tools/ktx/command.h

```diff
@@ -83,30 +146,164 @@ class Command {
+            format = OutputFormat::json_mini;
+        } else {
+            report.fatal_usage("Unsupported format: \"{}\"", formatStr);
+        }
+    }
+};
+
+struct OptionsSingleIn {
+    std::string inputFilepath;
+
+    void init(cxxopts::Options& opts) {
+        opts.add_options()
+                ("s,stdin", "Use stdin as the input file")
+                ("i,input-file", "The input file", cxxopts::value<std::string>(), "filepath");
+        opts.parse_positional("input-file");
+        opts.positional_help("<input-file>");
+    }
+
+    void process(cxxopts::Options&, cxxopts::ParseResult& args, Reporter& report) {
+        if (args.count("stdin") + args.count("input-file") == 0)
+            report.fatal_usage("Missing input file. Either <input-file> or <stdin> must be specified.");
+        if (args.count("stdin") + args.count("input-file") > 1)
+            report.fatal_usage("Conflicting options: Only one can be specified from <input-file> and <stdin>.");
+
+        // TODO Tools P4: Add support for stdin (To support '-' alias argv has to be scanned as cxxopts has no direct support for it)
+        if (args.count("stdin"))
+            inputFilepath = "-";
+        else
+            inputFilepath = args["input-file"].as<std::string>();
+    }
+};
+
+struct OptionsSingleInSingleOut {
+    std::string inputFilepath;
+    std::string outputFilepath;
+
+    void init(cxxopts::Options& opts) {
+        opts.add_options()
+                ("s,stdin", "Use stdin as the input file")
+                ("i,input-file", "The input file", cxxopts::value<std::string>(), "filepath")
+                ("o,output-file", "The output file", cxxopts::value<std::string>(), "filepath");
+        opts.parse_positional("input-file", "output-file");
+        opts.positional_help("<input-file> <output-file>");
+    }
+
+    void process(cxxopts::Options&, cxxopts::ParseResult& args, Reporter& report) {
+        if (args.count("stdin") + args.count("input-file") == 0)
+            report.fatal_usage("Missing input file. Either <input-file> or <stdin> must be specified.");
+        if (args.count("stdin") + args.count("input-file") > 1)
+            report.fatal_usage("Conflicting options: Only one can be specified from <input-file> and <stdin>.");
+
+        // TODO Tools P4: Add support for stdin (To support '-' alias argv has to be scanned as cxxopts has no direct support for it)
+        if (args.count("stdin"))
+            inputFilepath = "-";
+        else
+            inputFilepath = args["input-file"].as<std::string>();
+
+        outputFilepath = args["output-file"].as<std::string>();
+    }
+};
+
+struct OptionsMultiInSingleOut {
+    std::vector<std::string> inputFilepaths;
+    std::string outputFilepath;
+
+    void init(cxxopts::Options& opts) {
+        opts.add_options()
+                ("files", "Input/output files. Last file specified will be used as output", cxxopts::value<std::vector<std::string>>(), "<filepath>");
```

Comment:
cxxopts's positional arguments work like this, its a bit odd, but this positional arguments are declared just like the rest but it will be hidden from the usage list. So there is fact a hidden --files param that could also be used for positional arguments.

Reply:
I see. Thanks.